pan-domain-auth-verification/src/main/scala/com/gu/pandomainauth/PanDomain.scala

```diff
@@ -0,0 +1,53 @@
+package com.gu.pandomainauth
+
+import com.gu.pandomainauth.model._
+import com.gu.pandomainauth.service.{LegacyCookie, CookieUtils}
+
+
+object PanDomain {
+  /**
+   * Check the authentication status of the provided credentials by examining the signed cookie data.
+   */
+  def authStatus(cookieData: String, publicKey: String, validateUser: AuthenticatedUser => Boolean = guardianValidation): AuthenticationStatus = {
+    try {
+      val authedUser = CookieUtils.parseCookieData(cookieData, publicKey)
+      checkStatus(authedUser, validateUser)
+    } catch {
+      case e: Exception =>
+        InvalidCookie(e)
+    }
+  }
+
+  /**
+   * Cookie check that includes the legacy cookie. This is required during a period of transition
+   * between the old cookie and the new assymetric one.
+   */
+  def authStatusWithLegacyCheck(cookieData: String, publicKey: String, secret: String): AuthenticationStatus = {
```

Comment:
Yeah this probably is worth doing. I didn't duplicate all the tests for this temporary method but it's totally critical to the library's workings so I should probably just go ahead and do that for now.